### PR TITLE
add hpa api

### PR DESCRIPTION
**Pull request by richardxz**

### CODE REVIEW

File: pkg/apis/v1alpha/hpa/hpa.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+func getHpa(req *restful.Request, resp *restful.Response) {
+	hpa := req.PathParameter("horizontalpodautoscaler")
+	namespace := req.PathParameter("namespace")
+	client := client.NewK8sClient()
+
+	res, err := client.AutoscalingV1().HorizontalPodAutoscalers(namespace).Get(hpa, metaV1.GetOptions{})
+
+	if err != nil {
+		if errors.IsNotFound(err) {
+			resp.WriteHeaderAndEntity(http.StatusOK, nil)
```
Comment (wnxn) [inline pkg/apis/v1alpha/hpa/hpa.go]: The response status might be set as `not found` instead of `ok`.

Comment (richardxz) [reply pkg/apis/v1alpha/hpa/hpa.go]: this is frontend's need,  if return 404, the web page will not work properly

File: pkg/apis/v1alpha/hpa/hpa.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+		} else {
+			resp.WriteHeaderAndEntity(http.StatusInternalServerError, nil)
```
Comment (wnxn) [inline pkg/apis/v1alpha/hpa/hpa.go]: Shall we return error message ?

File: pkg/apis/v1alpha/hpa/hpa.go (hunk 1 above)
Comment (wnxn) [reply pkg/apis/v1alpha/hpa/hpa.go]: Ok, I understand what you mean.

File: pkg/apis/v1alpha/hpa/hpa.go (hunk 2 above)
Comment (richardxz) [reply pkg/apis/v1alpha/hpa/hpa.go]: don't need,  nobody care this

**Pull request merged**